### PR TITLE
feat : change FilterAdder api to take allowDuplicateFields

### PR DESCRIPTION
**Pull request by ccalbaugh**
- [x] override `uniqueFields` based on `allowDuplicateFields` passed in `SearchFilters` children

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Please assign someone to merge this PR, and optionally include people who should review.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>The changelog has not been updated. Please update the changelog.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>The version was not updated. Please update the version.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>You should add at least 1 more reviewer to the PR</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Branch being merged does not follow [Git Flow](http://nvie.com/posts/a-successful-git-branching-model/)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>You reduced the total lines of code! Awesome! :+1:</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3210104715a38f67dfe16d2aca4e0bdce2fed170
</p>

**Comment by stellarhoof** [comment]
We really waited and waited to release 2.0, I don't think this warrants a major bump. Can you make it non-breaking?

**Comment by ccalbaugh** [comment]
@stellarhoof @daedalus28 I can do whatever the team wants/needs, but this is a breaking change, I just want that to be clear.

**Comment by daedalus28** [comment]
As per our discussion, the idea is instead to add `allowDuplicateFields` in `SearchFilters` but keep `uniqueFields` at the `FilterAdder` level

File: src/FilterAdder.js [hunk 1]
```diff
@@ -14,12 +14,12 @@ let FilterAdder = ({
   node,
   path,
   fields,
-  uniqueFields,
+  allowDuplicateFields,
```
Comment (stellarhoof) [inline src/FilterAdder.js]: Well this _is_ a breaking change since `uniqueFields` is part of the public API

Comment (ccalbaugh) [reply src/FilterAdder.js]: Yeah I just reverted it back

File: src/SearchFilters.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ let LabelledList = ({ list, Component }) =>
 export let AddableFilterList = props => (
   <>
     <FilterList {...props} />
-    <FilterAdder {...props} uniqueFields />
+    <FilterAdder {...props} uniqueFields={!props.allowDuplicateFields} />
```
Comment (daedalus28) [inline src/SearchFilters.js]: I think you also need to thread this through search filters / search tree or whatever, right?

**Pull request merged**